lambdas/service/app.py

```diff
@@ -222,44 +222,154 @@ def openapi_raw():
                     body=raw_spec)
 
 
-health_spec = {
-    'responses': {
-        '200': {
-            'description': '200 response',
-            'content': {
-                'application/json': {
-                    'schema': {'$ref': '#/components/schemas/Empty'}
-                }
-            }
-        }
-    },
-    'tags': ['Auxiliary']
+health_up_key = {
+    'up': 'indicates with a boolean value if the service is reachable.\n',
+}
+
+health_service_keys = {
+    'elasticsearch': 'indicates whether service can establish a connection with Elasticsearch, '
+                     'ensuring that indexed documents can be access through Elasticsearch.\n',
+    'api_endpoints': 'shows a list of service API endpoints being requested by service, '
+                     'confirming that a client making a similar request will obtain a OK response.\n',
+    **health_up_key
+}
+
+health_all_keys = {
+    'queues': 'lists queues utilized by indexer to process transactions and index documents into '
+              'Elasticsearch. A queue is denoted as healthy if queue stats can be collected, stats are then '
+              'displayed for each available queue.\n',
+    'other_lambdas': 'represent other server-less applications used by Azul in order to make '
+                     'web service available, index is an example. Index processes data and indexes it into '
+                     'Elasticsearch. Indexed data is then available for service to search through via Elasticsearch. '
+                     'The status check for other lambdas is obtained by getting a response from their `health/basic` '
+                     'endpoints, ensuring that the application is reachable.\n',
+    'progress': 'is a more concrete version of __queues__. Progress indicates the remainder of '
+                'transactions and documents to be indexed. Which gives the client insight into the progress of '
+                'indexing.\n',
+    **health_service_keys
 }
 
 
+def health_docs_spec(keys: dict) -> str:
+    return ''.join(f'* __{k}__ {v} ' for k, v in keys.items())
+
+
+def health_example_spec(keys: dict, up: Optional[bool] = None):
+    example = {key: {} for key in keys.keys()}
+    example['up'] = up
+    return example
+
+
+def health_spec(health_keys):
+    return {
+        'responses': {
+            f'{200 if up else 503}': {
+                'description': f'Service component is {"" if up else "un"}reachable.'
```

Comment:
format_description

Reply:
@hannes-ucsc it is worth noting that I was unable to integrate unwrapping and documenting of the `health_keys` inside the `f` string. Doing so deformed the whole `description` object. Which is why I concatenated it.